tests-beta/ios/MASVS-STORAGE/MASTG-TEST-0x55-2.md

```diff
@@ -0,0 +1,25 @@
+---
+platform: ios
+title: Sensitive Data Present In Keyboard Cache
+id: MASTG-TEST-0x55-2
+type: [dynamic]
+weakness: MASWE-0053
+---
+
+## Overview
+
+This test checks whether the keyboard cache contains sensitive data from the target app.
+
+## Steps
```

Comment:
is this test really useful/meaningful/valuable?

cc @TheDauntless

Reply:
After thinking it over again, I am not sure about value of these tests:
*  Inspecting `/private/var/mobile/Library/Keyboard/en-dynamic.lm` isn't consistent. Typing a password with autocorrection doesn't consistently put it to the database. Therefore, it's even difficult to get a True Positive.
* Static search for `UITextAutocorrectionTypeNo` and `secureTextEntry` is also difficult if an app uses Storyboard. It'd be easier in SwiftUI but this is not a big percentage of apps yet. Even if it was possible for all apps, I am not sure if testing of usage of these APIs would be practical and sufficient.

In general, it's difficult to test for this weakness statically because we don't know what data is sensitive and where it's typed. Therefore, I suggest we:
1. Remove the test for testing `/private/var/mobile/Library/Keyboard`
2. Remove the static search for `UITextAutocorrectionTypeNo` and `secureTextEntry`
3. Create a dynamic test with Frida that intercepts text from all inputs that don't have `UITextAutocorrectionTypeNo` or `secureTextEntry`, and let the user review this list afterwards.